src/Detector/Node/NodeDetector.cs

```diff
@@ -233,14 +240,19 @@ private IEnumerable<FrameworkInfo> DetectFrameworkInfos(DetectorContext context)
                 // wild-card dependency
                 (bool isWildCardDependency, string wildCarddependencyName) = GetWildCardDependency(dependencyName);
 
-                if (monitoredDependencies.ContainsKey(dependencyName) || isWildCardDependency)
+                bool isValidDependency = monitoredDependencies.ContainsKey(dependencyName);
+                string frameworkName = isValidDependency ? monitoredDependencies[dependencyName] : "";
+                frameworkName = isWildCardDependency ? wildCarddependencyName : frameworkName; 
+                bool isNotDuplicateFramework = !(frameworksSet.Contains(frameworkName));
+                if (isValidDependency && isNotDuplicateFramework)
                 {
                     var frameworkInfo = new FrameworkInfo
                     {
-                        Framework = isWildCardDependency ? wildCarddependencyName : monitoredDependencies[dependencyName],
+                        Framework = frameworkName,
                         FrameworkVersion = dependency.Value.Value
                     };
                     detectedFrameworkResult.Add(frameworkInfo);
+                    frameworksSet.Add(frameworkName);
```

Comment:
nit: it may be worth looking to see if we can extract this logic in the `foreach` loop out into a separate method and reuse it between the dependencies and dev dependencies checks. It appears the only thing differing between the two cases are (1) iterating over `dependencies` vs. `devDependencies`, and (2) checking for the `dependencyName` in `monitoredDependencies` vs. `monitoredDevDependencies`, both cases we should be able to handle in a single method.

Reply:
I've added a work-item 1493329 to refactor later. I've added points to consider in the story description